plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerTypeMapping.java

```diff
@@ -23,20 +24,239 @@
+    private final LocalDateTime timeGapInVilnius = LocalDateTime.of(2018, 3, 25, 3, 17, 17);
+    private final LocalDateTime timeDoubledInVilnius = LocalDateTime.of(2018, 10, 28, 3, 33, 33, 333_000_000);
+
+    // minutes offset change since 1970-01-01, no DST
+    private final ZoneId kathmandu = ZoneId.of("Asia/Kathmandu");
+    private final LocalDateTime timeGapInKathmandu = LocalDateTime.of(1986, 1, 1, 0, 13, 7);
+
+    @BeforeClass
+    public void setUp()
+    {
+        checkIsGap(jvmZone, timeGapInJvmZone1);
+        checkIsGap(jvmZone, timeGapInJvmZone2);
+        checkIsDoubled(jvmZone, timeDoubledInJvmZone);
+
+        checkIsGap(vilnius, timeGapInVilnius);
+        checkIsDoubled(vilnius, timeDoubledInVilnius);
+
+        checkIsGap(kathmandu, timeGapInKathmandu);
+    }
+
+    @Test
+    public void testBasicTypes()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("boolean", "null", BOOLEAN, "CAST(NULL AS BOOLEAN)")
+                .addRoundTrip("boolean", "true", BOOLEAN)
+                .addRoundTrip("boolean", "false", BOOLEAN)
+                .addRoundTrip("bigint", "null", BIGINT, "CAST(NULL AS BIGINT)")
+                .addRoundTrip("bigint", "123456789012", BIGINT)
+                .addRoundTrip("integer", "null", INTEGER, "CAST(NULL AS INTEGER)")
+                .addRoundTrip("integer", "123456789", INTEGER)
+                .addRoundTrip("smallint", "null", SMALLINT, "CAST(NULL AS SMALLINT)")
+                .addRoundTrip("smallint", "32456", SMALLINT, "SMALLINT '32456'")
+                .addRoundTrip("tinyint", "null", TINYINT, "CAST(NULL AS TINYINT)")
+                .addRoundTrip("tinyint", "5", TINYINT, "TINYINT '5'")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_basic_types"));
+    }
+
+    @Test
+    public void testReal()
+    {
+        // we are not testing Nan/-Infinity/+Infinity as those are not supported by SQL Server
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS real)")
+                .addRoundTrip("real", "3.14", REAL, "REAL '3.14'")
+                .addRoundTrip("real", "3.1415927", REAL, "REAL '3.1415927'")
+                .execute(getQueryRunner(), sqlServerCreateAndInsert("test_real"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("real", "NULL", REAL, "CAST(NULL AS real)")
+                .addRoundTrip("real", "3.14", REAL, "REAL '3.14'")
+                .addRoundTrip("real", "3.1415927", REAL, "REAL '3.1415927'")
+                .execute(getQueryRunner(), trinoCreateAsSelect("test_real"));
+    }
+
+    @Test
+    public void testDouble()
+    {
+        // we are not testing Nan/-Infinity/+Infinity as those are not supported by SQL Server
+        SqlDataTypeTest.create()
+                .addRoundTrip("double precision", "NULL", DOUBLE, "CAST(NULL AS double)")
+                .addRoundTrip("double precision", "1.0E100", DOUBLE, "1.0E100")
```

Comment:
Add a test case with a fractional number

(add to `io.trino.plugin.postgresql.TestPostgreSqlTypeMapping#testDouble` too)